Firestore/core/src/model/value_util.cc

```diff
@@ -832,6 +846,20 @@ Message<google_firestore_v1_ArrayValue> DeepClone(
   return target;
 }
 
+Message<google_firestore_v1_MapValue> DeepClone(
+    const google_firestore_v1_MapValue& source) {
+  Message<google_firestore_v1_MapValue> target{source};
+  target->fields_count = source.fields_count;
+  target->fields = nanopb::MakeArray<google_firestore_v1_MapValue_FieldsEntry>(
+      source.fields_count);
+  for (pb_size_t i = 0; i < source.fields_count; ++i) {
+    target->fields[i].key = nanopb::MakeBytesArray(source.fields[i].key->bytes,
+                                                   source.fields[i].key->size);
+    target->fields[i].value = *DeepClone(source.fields[i].value).release();
+  }
+  return target;
+}
```

Comment:
This is a duplicate of the logic in DeepClone:

https://github.com/firebase/firebase-ios-sdk/blob/7b9125a2e99529e48a2e19dd6892c950e3dda58d/Firestore/core/src/model/value_util.cc#L772

can you call the existing DeepClone? (or refactor)

Reply:
https://github.com/firebase/firebase-ios-sdk/blob/7b9125a2e99529e48a2e19dd6892c950e3dda58d/Firestore/core/src/model/value_util.cc#L806-L818